### DISCUSSION TITLE
Query Frontend Responds 404 when using as Grafana Source

### DISCUSSION THREAD

**Discussion by m-wack** [grafana/mimir#2668] [Help and support]
Hello, not quite sure if this is an Issue with Mimir itself or Grafana, so please correct me if I'm wrong.
I have setup Mimir in Kubernetes using the following config:

alertmanager:
enable_api: false
ruler:
enable_api: false

I am using Mimir 2.2.0 and Grafana 9.0.1.

I am using Mimir as a Prometheus Datasource in Grafana with  
manageAlerts: false

However, when going into Explore in Grafana and selecting the Mimir source, Grafana does a Request to:
/api/datasources/4/resources/api/v1/rules

This results in a 404 Not Found Banner shown in Grafana. Everything is working though, I can do queries, I see stuff. It's just that this error pops up everytime. I noticed in the Changelog for Mimir v2.0 it says:

```
[CHANGE] Ruler: endpoints for listing configured rules now return HTTP status code 200 and an empty map when there are no rules instead of an HTTP 404 and plain text error message. The following endpoints are affected: #456

<prometheus-http-prefix>/config/v1/rules
<prometheus-http-prefix>/config/v1/rules/{namespace}
<prometheus-http-prefix>/rules (deprecated)
<prometheus-http-prefix>/rules/{namespace} (deprecated)
/api/v1/rules (deprecated)
/api/v1/rules/{namespace} (deprecated)
```

This does not seem to fix my problem though. Any ideas or tips for a fix?
Regards

**Comment by 56quarters** [comment]
There's a couple things going on here:
* All requests should be going through the Nginx reverse proxy that the helm chart sets up. It will route things to the appropriate backend service (distributor, query-frontend, ruler, etc)
* You've disabled the ruler API entirely, that's probably why you're getting a 404.

**Comment by m-wack** [comment]
Hi, thanks for your reply. We are using our own Nginx Reverse Proxy in front of it and are going directly through to the corresponding services, hence the nginx of the Chart is disabled.

I've disabled the ruler API, yes. However we do not need it and it's optional. So why does Grafana still query it when using the explore function? 
I am also using Loki, also with the ruler API disabled and with it I am not getting any errors.

**Reply by 56quarters** [reply]
> I've disabled the ruler API, yes. However we do not need it and it's optional. So why does Grafana still query it when using the explore function?

I'm not sure, you'd have to ask the Grafana team/project. My guess is that it's doing some sort of datasource "discovery" because it expects Mimir to support a ruler and it doesn't know anything about how you've configured your particular Mimir cluster.
